core/src/main/java/org/apache/accumulo/core/iterators/user/ScanYieldException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.iterators.user;
+
+import org.apache.accumulo.core.data.Key;
+
+/**
+ * This exception can be thrown at from a next or seek call on an iterator to allow other scans to get time slots. This mechanism is intended to avoid a set of
+ * scans to dominate all of the scan slots (readahead threads) and starve other scans out.
+ */
+public class ScanYieldException extends RuntimeException {
+  private final Key position;
+
+  public ScanYieldException(Key position) {
```

Comment:
I was attempting to get you past the incorrect statement of "For seek(), the start key of the Range passed to the Iterator".  The key will not be the start key of the range passed to the iterator.  Instead it will be the key where the seek left off in the underlying scan such that the next seek call will let it continue where it left off.